pages/apim/3.x/event-native-apim/event-native-apim-new-policy-execution-engine-evolution.adoc

```diff
@@ -0,0 +1,318 @@
+[[event-native-apim-new-policy-execution-engine-evolution]]
+= V3 and V4 (BETA) policy execution engines compared
+:page-sidebar: apim_3_x_sidebar
+:page-permalink: apim/3.x/event_native_apim_new_policy_execution_engine_evolution.html
+:page-folder: apim/event-native-apim
+:page-layout: apim3x
+
+[label label-version]#New in version 3.20.0#
+[label label-version]#BETA release#
+
+In this section you can learn about all the differences between the new (V4) execution engine mode and the existing V3 execution engine mode, as well as guidance on managing changes in system behavior and potential design and operational impact when switching to the new mode. The information is grouped by functional area.
+
+== Policy execution phases/scopes and execution order
+
+=== V3 mode
+
+In V3 mode, different execution scopes are required in order to indicate at which level a policy will work, as follows:
+
+* `REQUEST`: The policy only works on request headers. It never accesses the request body.
+* `REQUEST_CONTENT`: The policy works at request content level and can access the request body.
+* `RESPONSE`: The policy only works on response headers. It never accesses the response body.
+* `RESPONSE_CONTENT`: The policy works at response content level and can access the response body.
+
+As a result, all policies working on the body content are postponed to be executed after the policies working on headers. This leads to a different execution order than the one originally designed. This is shown in the following diagram:
+
+image:{% link /images/apim/3.x/event-native/event-native-api-management-execution-scopes-1.png %}[]
+
+=== New (V4) execution mode
+
+In the new (V4) execution mode, the `REQUEST_CONTENT` and `RESPONSE_CONTENT` scopes (which we now call 'phases') are no longer considered - all policies are executed in the exact order of design, regardless of whether they work on the content or not. This is shown in the following diagram:
+
+image:{% link /images/apim/3.x/event-native/event-native-api-management-execution-scopes-2.png %}[]
+
+=== Potential impact
+
+If you have designed your APIs with V3 execution engine mode ordering in mind and you link:{{ 'apim/3.x/event_native_apim_new_policy_execution_engine_activation.html' | relative_url }}[switch] from V3 to the new V4 execution engine mode, there may be execution behavior changes because the policy execution will be reordered to match the original design policy order.
```

Comment:
you cannot execute V3 api with V4 engine, because you will need to migrate it. However you can execute V3 api with jupiter engine which is not exactly the same engine.

Reply:
OK, I have clarified this paragraph.